### PR TITLE
Separate metrics poll and post loops

### PR DESCRIPTION
**Pull request by karitra**
Metrics exchange done via metrics.Hub structure.  Metrics gather loop done in frequent manner and accumulates moving average stats while post loop writes accumulated metrics to unicorn in long term periods.

Additionally:
+ Separate section for netlink config.
+ more metrics tests.

### CODE REVIEW

File: src/cocaine/burlak/burlak.py [hunk 1]
```diff
@@ -477,54 +479,67 @@ def subscribe_to_state_updates(self, unicorn):
...
+    def __init__(self, context, committed_state, hub, submitter, **kwargs):
+        super(MetricsSubmitter, self).__init__(context, **kwargs)
+        self._config = context.config
+        self._ci_state = committed_state
+        self._hub = hub
+        self._submitter = submitter
+
+    @gen.coroutine
+    def post_metrics(self):
+        while(self.should_run()):
```
Comment (shaitan) [inline src/cocaine/burlak/burlak.py]: You don't usually use `()` in such cases.

File: src/cocaine/burlak/burlak.py [hunk 2]
```diff
@@ -477,54 +479,67 @@ def subscribe_to_state_updates(self, unicorn):
...
         # TODO: update internal metrics
-        self.debug('fetching and updating workers stat took {:.3f}s', elapsed)
+        self.debug('fetching stat took {:.3f}s', elapsed)
 
     @gen.coroutine
     def poll_stats(self):
         while self.should_run():
             try:
-                to_sleep = self._config.metrics.poll_interval_sec
+                to_sleep = self._config.metrics.gather_interval_sec
```
Comment (shaitan) [inline src/cocaine/burlak/burlak.py]: Can we name them `poll_interval_sec` and `post_interval_sec`?

File: src/cocaine/burlak/config.py [hunk 3]
```diff
@@ -101,6 +108,21 @@ def make_sharding_config(d):
...
 
 
+def make_netlink_config(d):
+    Netlink = namedtuple('Netlink', [
+        'default_name',
+        'speed_mbits',
+    ])
+
+    default_name = d.get('default_name', Defaults.NETLINK_SPEED_MBITS)
+    speed_mbits = d.get('speed_mbits', Defaults.NETLINK_NAME)
```
Comment (shaitan) [inline src/cocaine/burlak/config.py]: `Defaults.NETLINK_NAME` is used for `speed_mbits` and `Defaults.NETLINK_SPEED_MBITS` for `default_name`

File: src/cocaine/burlak/config.py [hunk 4]
```diff
@@ -246,11 +268,23 @@ class Config(object):
                     'max': 2**16,
                     'required': False,
                 },
+                'post_interval': {
```
Comment (shaitan) [inline src/cocaine/burlak/config.py]: According to the other code, it should be called `poll_interval` and why is it without `_sec`?

File: src/cocaine/burlak/metrics/hub.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @property
+    def system(self):
+        return self._system
+
+    @system.setter
+    def system(self, s):
+        self._system = s
+
+    @property
+    def applications(self):
```
Comment (shaitan) [inline src/cocaine/burlak/metrics/hub.py]: Why do you need separated getters and setters? Why don't use just make `self._applications` -> `self.application`?

File: src/cocaine/burlak/metrics/procfs.py [hunk 6]
```diff
@@ -381,6 +385,11 @@ def read(self):
             drx = int(NET_TICKS_PER_SEC * (net2.rx - net1.rx))
             dtx = int(NET_TICKS_PER_SEC * (net2.tx - net1.tx))
 
+            rates = self.rates(iface)
```
Comment (shaitan) [inline src/cocaine/burlak/metrics/procfs.py]: I can't find where `self._reates` is updated. It shouldn't be or should?

File: src/cocaine/burlak/metrics/system.py [hunk 7]
```diff
@@ -17,18 +17,15 @@ class SystemMetrics(LoggerMixin):
     def __init__(self, context, **kwargs):
         super(SystemMetrics, self).__init__(context, **kwargs)
 
-        self._config = context.config
+        conf = context.config
+        self._config = conf
```
Comment (shaitan) [inline src/cocaine/burlak/metrics/system.py]: `self._config` added but isn't used

File: src/cocaine/burlak/burlak.py (hunk 1 above)
Comment (karitra) [reply src/cocaine/burlak/burlak.py]: It is used in tests mocking where it is possible to run loop specified number of times.

File: src/cocaine/burlak/metrics/procfs.py (hunk 6 above)
Comment (karitra) [reply src/cocaine/burlak/metrics/procfs.py]: Yep, it should be, fixed.

File: src/cocaine/burlak/defaults.py [hunk 8]
```diff
@@ -19,7 +19,8 @@ class Defaults(object):
 
     METRICS_PATH = '/darkvoice/metrics'
     METRICS_ENABLED = False
-    METRICS_POLL_INTERVAL_SEC = 10
+    METRICS_POST_INTERVAL_SEC = 20
+    METRICS_POLL_INTERVAL_SEC = 5
```
Comment (shaitan) [inline src/cocaine/burlak/defaults.py]: Why are these defaults chosen?

**Pull request merged**